src/components/drawer/drawer.component.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+
+import {
+  StyledDrawerWrapper,
+  StyledDrawerContent,
+  StyledIconButton,
+  StyledDrawerChildren,
+  StyledDrawerSidebar
+} from './drawer.style';
+
+const Drawer = ({
+  defaultIsExpanded,
+  expanded,
+  onChange,
+  children,
+  expandedSettings,
+  sidebar,
+  ...props
+}) => {
+  const isControlled = expanded !== undefined;
+  const [isExpandedInternal, setIsExpandedInternal] = useState(defaultIsExpanded || false);
+  const [firstOpen, setFirstOpen] = useState(true);
+  const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const toggleDrawer = useCallback((ev) => {
+    if (!isControlled) setIsExpandedInternal(!isExpanded);
+    if (onChange) onChange(ev, !isExpanded);
+    setFirstOpen(false);
+  }, [isControlled, isExpanded, onChange]);
+
+  const closedClass = firstOpen ? '' : 'closed';
+
+  return (
+    <StyledDrawerWrapper
+      isExpanded={ isExpanded }
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedSettings ? expandedSettings.maxWidth : '40%' }
```

Comment:
I think we should change this interface. The DLS specified a few explicit width options. 
See other suggestions below for complete example.
```suggestion
        expandedWidth={ expandedWidth }
```

Reply:
fixed